dev/DynamicDependency/M.AM.DD.PackageDependency.cpp

```diff
@@ -0,0 +1,124 @@
+        return Create(tokenUser->User.Sid, packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, mddLifetimeArtifact, mddOptions);
+    }
+
+    winrt::PackageDependency PackageDependency::Create(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion, winrt::CreatePackageDependencyOptions const& options)
+    {
+        auto tokenUser{ wil::get_token_information<TOKEN_USER>(GetCurrentThreadEffectiveToken()) };
+        const auto mddMinVersion{ ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion) };
+        const auto mddArchitectures{ ::Microsoft::ApplicationModel::DynamicDependency::ToArchitectures(options.Architectures()) };
+        const auto mddLifetimeKind{ ::Microsoft::ApplicationModel::DynamicDependency::ToLifetimeKind(options.LifetimeArtifactKind()) };
+        const auto mddLifetimeArtifact{ options.LifetimeArtifact().c_str() };
+        const auto mddOptions{ ::Microsoft::ApplicationModel::DynamicDependency::ToCreateOptions(options) };
+        return Create(tokenUser->User.Sid, packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, mddLifetimeArtifact, mddOptions);
+    }
+
+    winrt::PackageDependency PackageDependency::CreateForSystem(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion, winrt::CreatePackageDependencyOptions const& options)
+    {
+        const auto mddMinVersion{ ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion) };
+        const auto mddArchitectures{ ::Microsoft::ApplicationModel::DynamicDependency::ToArchitectures(options.Architectures()) };
+        const auto mddLifetimeKind{ ::Microsoft::ApplicationModel::DynamicDependency::ToLifetimeKind(options.LifetimeArtifactKind()) };
+        const auto mddLifetimeArtifact{ options.LifetimeArtifact().c_str() };
+        auto mddOptions{ ::Microsoft::ApplicationModel::DynamicDependency::ToCreateOptions(options, MddCreatePackageDependencyOptions::ScopeIsSystem) };
+        return Create(nullptr, packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, mddLifetimeArtifact, mddOptions);
+    }
+
+    hstring PackageDependency::Id()
+    {
+        return m_id;
+    }
+
+    void PackageDependency::Delete()
+    {
+        MddDeletePackageDependency(m_id.c_str());
+    }
+
+    winrt::PackageDependencyContext PackageDependency::Add()
+    {
+        const auto rank{ MDD_PACKAGE_DEPENDENCY_RANK_DEFAULT };
+        const auto mddOptions{ MddAddPackageDependencyOptions::None };
+        wil::unique_package_dependency_context packageDependencyContext;
+        winrt::check_hresult(MddAddPackageDependency(m_id.c_str(), rank, mddOptions, &packageDependencyContext, nullptr));
+        auto context{ winrt::make<implementation::PackageDependencyContext>(std::move(packageDependencyContext.get())) };
+        packageDependencyContext.release();
```

Comment:
Why do you need an explicit release?

Reply:
That's a really good question. Errr, because it was one of the first things I wrote and when code morphed I missed removing the obsolete .release? <sheepish grin>

Removed. Thanks!